indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,6 +161,8 @@ public IndexTask(
     super(
         // _not_ the version, just something uniqueish
         makeId(id, ingestionSchema),
+        String.format("index_%s", makeDataSource(ingestionSchema)),
```

Comment:
The default thing should definitely to be have groupId equal taskId. Groups are for lock-sharing; tasks with the same groupId can share locks. Realtime tasks need to do this, but IndexTasks don't need to and shouldn't be sharing locks.